heartpoints.sh

```diff
@@ -44,6 +59,70 @@ heartpoints_localDev() {
     error_and_exit "localDev is deprecated. For server side development, use ./hp serverDev or for client side development use ./hp clientDev"
 }
 
+string_toLower() { local stringToConvertToAllLowercase=$1
+    echo "$stringToConvertToAllLowercase" | tr '[:upper:]' '[:lower:]'
+}
+
+string_firstN() { local sourceString=$1; local lengthNOfDesiredCharsFromTheLeft=$2
+    echo "${sourceString}" | cut -c 1-$lengthNOfDesiredCharsFromTheLeft
+}
+
+string_everythingAfterN() { local sourceString=$1; local indexThatWillBecomeFirstCharOfReturnedString=$2
+    echo "${sourceString:$indexThatWillBecomeFirstCharOfReturnedString}"
+}
+
+string_everythingAfterChar() { local sourceString=$1; local delimitingCharacter=$2;
+    cut -d "${delimitingCharacter}" -f 2 <<< "$sourceString"
+}
+
+git_safeBranchNameFromSentence() { local sentence=$1
+    local lowercased="$(string_toLower "${sentence}")"
+    local maxLength="50"
+    local trimmed="$(string_firstN "${lowercased}" $maxLength)"
+    local spacesReplacedWithDashes="${trimmed// /-}"
+    local withoutPoundSignOrLeadingSpace="$(string_everythingAfterChar "${spacesReplacedWithDashes}" "#")"
```

Comment:
it seems that this function is meant to be used as part of `git_safeBranchNameForIssueId`, given the "everything after `#`" rule. What do you think about merging the two rather than having two separate functions?

Reply:
renamed it to explicate its dependence and particularity around github issue description